include/maxminddb.h

```diff
@@ -135,6 +135,7 @@ typedef struct MMDB_entry_data_s {
 typedef struct MMDB_entry_data_list_s {
     MMDB_entry_data_s entry_data;
     struct MMDB_entry_data_list_s *next;
+    bool head;
```

Comment:
Another approach would have been to keep the pool around. We wouldn't have to traverse the whole list in that case, but I suppose it doesn't matter too much.

Reply:
Hmm, are you thinking we could stick a pointer to the pool in `MMDB_entry_data_list_s`? I guess we could use a `void*` (with the idea of not exposing `data-pool.h` for no reason).

Reply:
Yeah, something like that. It would eliminate some of the work when freeing.